app/controllers/ExperimentsController.php

```diff
@@ -143,11 +143,15 @@
         try {
             $Entity->setId($_POST['id']);
             $Entity->canOrExplode('write');
-            $ts = new TrustedTimestamps(new Config(), new Teams($_SESSION['team_id']), $Entity);
-            if ($ts->timeStamp()) {
-                echo json_encode(array(
+            if ($Entity->isTimestampable()) {
+                $ts = new TrustedTimestamps(new Config(), new Teams($_SESSION['team_id']), $Entity);
+                if ($ts->timeStamp()) {
+                    echo json_encode(array(
                     'res' => true
                 ));
+                }
+            } else {
+                throw new Exception(_('This experiment cannot be timestamped!'));
```

Comment:
This is called by ajax, so throwing an exception here is useless. It's better to reply a json message with res:false and msg:errormsg.